packages/rabbitmq/src/rabbitmq.module.ts

```diff
@@ -122,9 +122,11 @@ export class RabbitMQModule
   async onApplicationShutdown() {
     this.logger.verbose('Closing AMQP Connections');
 
-    for (const connection of this.connectionManager.getConnections()) {
-      await connection.managedConnection.close();
-    }
+    await Promise.all(
+      this.connectionManager
+        .getConnections()
+        .map((connection) => connection.managedConnection.close())
```

Comment:
`.map((connection) => connection.managedConnection.close)` will be better because just passing the reference will make sure that the promise handler will invoke and await but Its just a suggestion